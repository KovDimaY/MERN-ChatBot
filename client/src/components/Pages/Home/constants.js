import {
  flagES,
  flagRU,
  flagUA,
  flagUS,
  hobbyNumismatics,
  hobbyPhotography,
  hobbyProgramming,
  hobbyRubik,
  hobbySports,
} from 'images';

export const aboutMeDescription = [
  `I am a very passionate software developer with a mathematical
  background who has tried working with many different languages
  including Pascal, FORTRAN, Python, C/C++, C#, Java and Javascript.
  Now I am focused on web development with Node and React and I love it!`,
  `I am a very methodological and disciplined person,
  I like when things are done in time and with the best
  possible quality. I love test coverage, CI and strict
  ESLint rules but also I like to experiment and to try
  new approaches when I see that something could be improved.
  I am very hardworking and a bit crazy about learning - I
  cannot imagine my life without Coursera, Udemy and GitHub.`,
  `I believe in productive teamwork and I consider myself a good team
  player who can have fun and support others but also to be strict and
  demanding when needed. I am always trying to find something positive
  to keep an optimistic attitude and to be happy. I use sport and photography
  when I need to distract and relax.`,
];

export const languages = [
  {
    image: flagUA,
    name: 'Українська',
    level: 'Native or bilingual proficiency',
    id: 'lnguage-1',
  },
  {
    image: flagRU,
    name: 'Русский',
    level: 'Native or bilingual proficiency',
    id: 'lnguage-2',
  },
  {
    image: flagUS,
    name: 'English',
    level: 'Full professional proficiency',
    id: 'lnguage-3',
  },
  {
    image: flagES,
    name: 'Español',
    level: 'Professional working proficiency',
    id: 'lnguage-4',
  },
];

export const hobbies = [
  {
    image: hobbyProgramming,
    name: 'Programming',
    description: `Even though I am working as a full time programmer,
      my biggest hobby is still programming. Lastly, I spend most of my
      free time doing some programming or learning about it. My passion
      started in the university, where I learned creating different games
      in C++, C# and Python. Now I enjoy a lot creating different web
      applications, especially if they can be helpful or interesting
      for other people. For the moment, the most pleasure I have working
      with Javascript, more precisely with Node and React.`,
    id: 'hobby-1',
    storageName: 'programming',
  },
  {
    image: hobbySports,
    name: 'Sports',
    description: `I love doing sports, especially team playing games like
      football, volleyball or basketball. While studying, I was playing
      quite seriously - I was even a captain of the faculty team in the
      university. Also I was very dedicated for several years to
      bodybuilding and powerlifting and I reached significant results,
      gaining more than 30 kg of body mass and good strength metrics.
      Also I enjoy playing table tennis, chess, billiard, running, cycling,
      swimming and other activities. Sport helps to keep fit my body and mind
      and also it is a great fun.`,
    id: 'hobby-2',
    storageName: 'sports',
  },
  {
    image: hobbyRubik,
    name: 'Puzzles',
    description: `I like different puzzles and particularly I have a big
      interest for Rubik’s cubes. I have a collection of different sizes
      from 2x2 to 10x10 and my dream is to get a cube of 17x17. I do not
      really like speed cubing, I prefer solving them slowly and without
      pressure, normally it helps me to relax and to distract my brain from
      possible stresses. For example, to solve the cube of size 10x10 I normally
      spend 3-4 hours.`,
    id: 'hobby-3',
    storageName: 'puzzles',
  },
  {
    image: hobbyNumismatics,
    name: 'Numismatics',
    description: `From a very childhood I like to collect different things
      and my biggest and the most important collection is my collection of coins.
      I have around 2000 different coins from different countries and this number
      is growing every year. Most of them I got while traveling or from my friends.
      I do not have some super special coins, my idea is just to collect as many
      different coins as I can.`,
    id: 'hobby-4',
    storageName: 'numismatics',
  },
  {
    image: hobbyPhotography,
    name: 'Photography',
    description: `I love making pictures and videos. Even though I have never had
      some big achievements in photography, I enjoy the process of getting a perfect
      shot, including editing and retouching. My favourite kind of photography
      is portrait. Making pictures of people is the most enjoyable for me but,
      also, I like to make pictures of animals and nature. I love when the picture
      is full of bright colours especially green, blue, yellow and red. I have a dream
      to make a photo exhibition and to make a short movie some day.`,
    id: 'hobby-5',
    storageName: 'photography',
  },
];
