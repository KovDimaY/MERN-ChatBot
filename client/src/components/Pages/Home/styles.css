.home-page-container {
  padding-bottom: 100px;
  max-width: 800px;
  margin: auto;
}
.home-page-container strong {
  font-weight: 600;
}
.home-page-container .basic-info-section h4 {
  margin: 20px 0 0 0;
}
.home-page-container .section {
  padding: 10px 25px;
}
.home-page-container .basic-info-section {
  margin-top: 25px;
}
.home-page-container .basic-info-section .left-column {
  text-align: center;
}
.home-page-container .basic-info-section .right-column {
  max-width: 300px;
}
.home-page-container .my-profile-photo {
  width: 250px;
  border-radius: 5px;
  box-shadow: 1px 1px 10px grey;
}
.home-page-container .basic-info-section .job-title {
  font-size: 20px;
  color: gray;
  margin: 5px 0 20px 0;
}
.home-page-container .section-title {
  font-size: 28px;
  margin: 0;
}
.home-page-container .summary-section p {
  text-align: justify;
  text-indent: 20px;
}
.home-page-container .languages-wrapper {
  margin-top: 25px;
}

@media (max-width: 750px) {
  .home-page-container .basic-info-section {
    font-size: 12px;
  }
  .home-page-container .basic-info-section h4 {
    font-size: 22px;
    margin: 0;
    font-weight: 600;
  }
  .home-page-container .basic-info-section .job-title {
    font-size: 16px;
    margin-bottom: 10px;
  }
  .home-page-container .basic-info-section p {
    font-size: 12px;
    margin: 5px 0;
  }
  .home-page-container .my-profile-photo {
    width: 150px;
  }
}

@media (max-width: 600px) {
  .home-page-container .basic-info-section {
    flex-direction: column;
    justify-content: center;
  }
  .home-page-container .basic-info-section .left-column {
    margin: 0;
  }
  .home-page-container .my-profile-photo {
    width: 200px;
  }
  .home-page-container .basic-info-section .right-column {
    text-align: center;
    max-width: none;
    margin-top: 25px;
  }
  .home-page-container .section-title {
    text-align: center;
    font-size: 24px;
  }
}
