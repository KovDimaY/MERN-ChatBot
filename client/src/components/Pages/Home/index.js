import React from 'react';
import PropTypes from 'prop-types';

/* eslint-disable react/no-unused-prop-types */

import { getAgeByBirthdate, showHiddenText } from 'utils/common';
import { myProfilePicture } from 'images';
import { aboutMeDescription, languages, hobbies } from './constants';

import EmptyView from '../../Common/EmptyView';
import LanguageItem from '../../Common/LanguageItem';
import HobbyItem from '../../Common/HobbyItem';
import ProgressBar from '../../Common/ProgressBar';

import './styles.css';

const Home = props => {
  const renderText = (array, show = true) =>
    array.map(item => <p key={item}>{showHiddenText(item, show)}</p>);

  const renderLanguages = () =>
    languages.map(item => (
      <LanguageItem {...item} key={item.id} discovered={props.languages} />
    ));

  const renderHobbies = () =>
    hobbies.map(item => (
      <HobbyItem
        {...item}
        key={item.id}
        discovered={props.hobbies.discovered}
        discoveredDetails={props.hobbies[item.storageName]}
      />
    ));

  const renderName = () => {
    if (props.name) {
      return 'Dmytro Kovalenko';
    }

    return (
      <React.Fragment>
        <strong>Name: </strong> {showHiddenText('Dmytro', false)}
      </React.Fragment>
    );
  };

  const renderPosition = () => {
    if (props.position) {
      return 'Full Stack JS Developer';
    }

    return (
      <React.Fragment>
        <strong>Profession: </strong> {showHiddenText('JS Developer', false)}
      </React.Fragment>
    );
  };

  const renderLocation = () => {
    if (props.location) {
      return (
        <a
          href="https://www.google.es/maps?q=41.382207,2.140318"
          target="_blank"
          rel="noopener noreferrer"
        >
          Sants Estaci√≥, Barcelona, Spain
        </a>
      );
    }

    return (
      <React.Fragment>
        <strong>Location: </strong>{' '}
        {showHiddenText('Sants Estacio, Barcelona', false)}
      </React.Fragment>
    );
  };

  const renderContactInfo = () => {
    if (props.contacts) {
      return (
        <React.Fragment>
          Contact me in{' '}
          <a
            href="https://www.linkedin.com/in/kovalenkodmytro/"
            target="_blank"
            rel="noopener noreferrer"
          >
            LinkedIn
          </a>
        </React.Fragment>
      );
    }

    return showHiddenText('Contact me in LinkedIn', false);
  };

  const renderBasicInfo = () => (
    <div className="basic-info-section section row">
      <div className="left-column col s12 m6">
        <img src={myProfilePicture} alt="Me" className="my-profile-photo" />
      </div>
      <div className="right-column col s12 m6">
        <h4 className="name-lastname">{renderName()}</h4>
        <p className="job-title">{renderPosition()}</p>
        <p className="location">{renderLocation()}</p>
        <p>
          <strong>Contact info: </strong>
          {renderContactInfo()}
        </p>
        <p>
          <strong>Age: </strong>
          {showHiddenText(
            `${getAgeByBirthdate('1992-05-20')} years old`,
            props.age
          )}
        </p>
        <p>
          <strong>Nationality: </strong>
          {showHiddenText('Ukraine', props.nationality)}
        </p>
      </div>
    </div>
  );

  const renderInfo = () => (
    <React.Fragment>
      <ProgressBar percent={props.discoveryPercent} />
      {renderBasicInfo()}

      <div className="summary-section section">
        <h3 className="section-title">About me:</h3>
        {renderText(aboutMeDescription, props.aboutMe)}
      </div>

      <div className="languages-section section">
        <h3 className="section-title">Languages I speak:</h3>
        <div className="languages-wrapper row">{renderLanguages()}</div>
      </div>

      <div className="hobbies-section section">
        <h3 className="section-title">My hobbies:</h3>
        <div className="hobbies-wrapper">{renderHobbies()}</div>
      </div>
    </React.Fragment>
  );

  const renderEmptyView = () => (
    <EmptyView section="home">
      <p>Welcome to my new personal project!</p>
      <p>
        Ask questions as you are speaking to me and try to populate the website
        with the info my bot will provide. Every page will have a progress bar
        which indicates how much info you have already recovered - so you will
        know where to stop asking about some particular topic.
      </p>
      <p>
        For example,{' '}
        <b>
          {'"'}What is your name?{'"'}
        </b>{' '}
        sounds like a good question to start, but it can be something else too,
        just use your imagination ;)
      </p>
    </EmptyView>
  );

  return (
    <div className="home-page-container">
      {props.discovered ? renderInfo() : renderEmptyView()}
    </div>
  );
};

Home.propTypes = {
  discovered: PropTypes.bool.isRequired,
  name: PropTypes.bool.isRequired,
  position: PropTypes.bool.isRequired,
  location: PropTypes.bool.isRequired,
  contacts: PropTypes.bool.isRequired,
  age: PropTypes.bool.isRequired,
  nationality: PropTypes.bool.isRequired,
  aboutMe: PropTypes.bool.isRequired,
  languages: PropTypes.bool.isRequired,
  hobbies: PropTypes.object.isRequired,
  discoveryPercent: PropTypes.number.isRequired,
};

export default Home;
