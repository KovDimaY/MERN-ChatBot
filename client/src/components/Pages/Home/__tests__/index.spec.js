import React from 'react';
import { create } from 'react-test-renderer';

import Home from '../../Home';

jest.mock('components/Common/EmptyView', () => 'EmptyView');
jest.mock('components/Common/ProgressBar', () => 'ProgressBar');
jest.mock('components/Common/LanguageItem', () => 'LanguageItem');
jest.mock('components/Common/HobbyItem', () => 'HobbyItem');
jest.mock('utils/common', () => ({
  getAgeByBirthdate: jest.fn(() => 26),
  showHiddenText: jest.fn(() => '***'),
}));

const mockComponent = props => <Home {...props} />;

describe('components/Pages/<Home />', () => {
  it('should render component with empty props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with discovered but all hidden', () => {
    const props = {
      discovered: true,
      discoveryPercent: 50,
      hobbies: {},
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with all discovered', () => {
    const props = {
      discovered: true,
      discoveryPercent: 100,
      name: true,
      position: true,
      location: true,
      contacts: true,
      age: true,
      nationality: true,
      aboutMe: true,
      languages: true,
      hobbies: {
        discovered: true,
        programming: true,
        sports: true,
        puzzles: true,
        numismatics: true,
        photography: true,
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
