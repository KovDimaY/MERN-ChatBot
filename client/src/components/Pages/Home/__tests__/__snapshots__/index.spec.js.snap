// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Pages/<Home /> should render component with all discovered 1`] = `
<div
  className="home-page-container"
>
  <ProgressBar
    percent={100}
  />
  <div
    className="basic-info-section section row"
  >
    <div
      className="left-column col s12 m6"
    >
      <img
        alt="Me"
        className="my-profile-photo"
        src="test-file-stub"
      />
    </div>
    <div
      className="right-column col s12 m6"
    >
      <h4
        className="name-lastname"
      >
        Dmytro Kovalenko
      </h4>
      <p
        className="job-title"
      >
        Full Stack JS Developer
      </p>
      <p
        className="location"
      >
        <a
          href="https://www.google.es/maps?q=41.382207,2.140318"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sants Estació, Barcelona, Spain
        </a>
      </p>
      <p>
        <strong>
          Contact info: 
        </strong>
        Contact me in
         
        <a
          href="https://www.linkedin.com/in/kovalenkodmytro/"
          rel="noopener noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </p>
      <p>
        <strong>
          Age: 
        </strong>
        ***
      </p>
      <p>
        <strong>
          Nationality: 
        </strong>
        ***
      </p>
    </div>
  </div>
  <div
    className="summary-section section"
  >
    <h3
      className="section-title"
    >
      About me:
    </h3>
    <p>
      ***
    </p>
    <p>
      ***
    </p>
    <p>
      ***
    </p>
  </div>
  <div
    className="languages-section section"
  >
    <h3
      className="section-title"
    >
      Languages I speak:
    </h3>
    <div
      className="languages-wrapper row"
    >
      <LanguageItem
        discovered={true}
        id="lnguage-1"
        image="test-file-stub"
        level="Native or bilingual proficiency"
        name="Українська"
      />
      <LanguageItem
        discovered={true}
        id="lnguage-2"
        image="test-file-stub"
        level="Native or bilingual proficiency"
        name="Русский"
      />
      <LanguageItem
        discovered={true}
        id="lnguage-3"
        image="test-file-stub"
        level="Full professional proficiency"
        name="English"
      />
      <LanguageItem
        discovered={true}
        id="lnguage-4"
        image="test-file-stub"
        level="Professional working proficiency"
        name="Español"
      />
    </div>
  </div>
  <div
    className="hobbies-section section"
  >
    <h3
      className="section-title"
    >
      My hobbies:
    </h3>
    <div
      className="hobbies-wrapper"
    >
      <HobbyItem
        description="Even though I am working as a full time programmer,
      my biggest hobby is still programming. Lastly, I spend most of my
      free time doing some programming or learning about it. My passion
      started in the university, where I learned creating different games
      in C++, C# and Python. Now I enjoy a lot creating different web
      applications, especially if they can be helpful or interesting
      for other people. For the moment, the most pleasure I have working
      with Javascript, more precisely with Node and React."
        discovered={true}
        discoveredDetails={true}
        id="hobby-1"
        image="test-file-stub"
        name="Programming"
        storageName="programming"
      />
      <HobbyItem
        description="I love doing sports, especially team playing games like
      football, volleyball or basketball. While studying, I was playing
      quite seriously - I was even a captain of the faculty team in the
      university. Also I was very dedicated for several years to
      bodybuilding and powerlifting and I reached significant results,
      gaining more than 30 kg of body mass and good strength metrics.
      Also I enjoy playing table tennis, chess, billiard, running, cycling,
      swimming and other activities. Sport helps to keep fit my body and mind
      and also it is a great fun."
        discovered={true}
        discoveredDetails={true}
        id="hobby-2"
        image="test-file-stub"
        name="Sports"
        storageName="sports"
      />
      <HobbyItem
        description="I like different puzzles and particularly I have a big
      interest for Rubik’s cubes. I have a collection of different sizes
      from 2x2 to 10x10 and my dream is to get a cube of 17x17. I do not
      really like speed cubing, I prefer solving them slowly and without
      pressure, normally it helps me to relax and to distract my brain from
      possible stresses. For example, to solve the cube of size 10x10 I normally
      spend 3-4 hours."
        discovered={true}
        discoveredDetails={true}
        id="hobby-3"
        image="test-file-stub"
        name="Puzzles"
        storageName="puzzles"
      />
      <HobbyItem
        description="From a very childhood I like to collect different things
      and my biggest and the most important collection is my collection of coins.
      I have around 2000 different coins from different countries and this number
      is growing every year. Most of them I got while traveling or from my friends.
      I do not have some super special coins, my idea is just to collect as many
      different coins as I can."
        discovered={true}
        discoveredDetails={true}
        id="hobby-4"
        image="test-file-stub"
        name="Numismatics"
        storageName="numismatics"
      />
      <HobbyItem
        description="I love making pictures and videos. Even though I have never had
      some big achievements in photography, I enjoy the process of getting a perfect
      shot, including editing and retouching. My favourite kind of photography
      is portrait. Making pictures of people is the most enjoyable for me but,
      also, I like to make pictures of animals and nature. I love when the picture
      is full of bright colours especially green, blue, yellow and red. I have a dream
      to make a photo exhibition and to make a short movie some day."
        discovered={true}
        discoveredDetails={true}
        id="hobby-5"
        image="test-file-stub"
        name="Photography"
        storageName="photography"
      />
    </div>
  </div>
</div>
`;

exports[`components/Pages/<Home /> should render component with discovered but all hidden 1`] = `
<div
  className="home-page-container"
>
  <ProgressBar
    percent={50}
  />
  <div
    className="basic-info-section section row"
  >
    <div
      className="left-column col s12 m6"
    >
      <img
        alt="Me"
        className="my-profile-photo"
        src="test-file-stub"
      />
    </div>
    <div
      className="right-column col s12 m6"
    >
      <h4
        className="name-lastname"
      >
        <strong>
          Name: 
        </strong>
         
        ***
      </h4>
      <p
        className="job-title"
      >
        <strong>
          Profession: 
        </strong>
         
        ***
      </p>
      <p
        className="location"
      >
        <strong>
          Location: 
        </strong>
         
        ***
      </p>
      <p>
        <strong>
          Contact info: 
        </strong>
        ***
      </p>
      <p>
        <strong>
          Age: 
        </strong>
        ***
      </p>
      <p>
        <strong>
          Nationality: 
        </strong>
        ***
      </p>
    </div>
  </div>
  <div
    className="summary-section section"
  >
    <h3
      className="section-title"
    >
      About me:
    </h3>
    <p>
      ***
    </p>
    <p>
      ***
    </p>
    <p>
      ***
    </p>
  </div>
  <div
    className="languages-section section"
  >
    <h3
      className="section-title"
    >
      Languages I speak:
    </h3>
    <div
      className="languages-wrapper row"
    >
      <LanguageItem
        id="lnguage-1"
        image="test-file-stub"
        level="Native or bilingual proficiency"
        name="Українська"
      />
      <LanguageItem
        id="lnguage-2"
        image="test-file-stub"
        level="Native or bilingual proficiency"
        name="Русский"
      />
      <LanguageItem
        id="lnguage-3"
        image="test-file-stub"
        level="Full professional proficiency"
        name="English"
      />
      <LanguageItem
        id="lnguage-4"
        image="test-file-stub"
        level="Professional working proficiency"
        name="Español"
      />
    </div>
  </div>
  <div
    className="hobbies-section section"
  >
    <h3
      className="section-title"
    >
      My hobbies:
    </h3>
    <div
      className="hobbies-wrapper"
    >
      <HobbyItem
        description="Even though I am working as a full time programmer,
      my biggest hobby is still programming. Lastly, I spend most of my
      free time doing some programming or learning about it. My passion
      started in the university, where I learned creating different games
      in C++, C# and Python. Now I enjoy a lot creating different web
      applications, especially if they can be helpful or interesting
      for other people. For the moment, the most pleasure I have working
      with Javascript, more precisely with Node and React."
        id="hobby-1"
        image="test-file-stub"
        name="Programming"
        storageName="programming"
      />
      <HobbyItem
        description="I love doing sports, especially team playing games like
      football, volleyball or basketball. While studying, I was playing
      quite seriously - I was even a captain of the faculty team in the
      university. Also I was very dedicated for several years to
      bodybuilding and powerlifting and I reached significant results,
      gaining more than 30 kg of body mass and good strength metrics.
      Also I enjoy playing table tennis, chess, billiard, running, cycling,
      swimming and other activities. Sport helps to keep fit my body and mind
      and also it is a great fun."
        id="hobby-2"
        image="test-file-stub"
        name="Sports"
        storageName="sports"
      />
      <HobbyItem
        description="I like different puzzles and particularly I have a big
      interest for Rubik’s cubes. I have a collection of different sizes
      from 2x2 to 10x10 and my dream is to get a cube of 17x17. I do not
      really like speed cubing, I prefer solving them slowly and without
      pressure, normally it helps me to relax and to distract my brain from
      possible stresses. For example, to solve the cube of size 10x10 I normally
      spend 3-4 hours."
        id="hobby-3"
        image="test-file-stub"
        name="Puzzles"
        storageName="puzzles"
      />
      <HobbyItem
        description="From a very childhood I like to collect different things
      and my biggest and the most important collection is my collection of coins.
      I have around 2000 different coins from different countries and this number
      is growing every year. Most of them I got while traveling or from my friends.
      I do not have some super special coins, my idea is just to collect as many
      different coins as I can."
        id="hobby-4"
        image="test-file-stub"
        name="Numismatics"
        storageName="numismatics"
      />
      <HobbyItem
        description="I love making pictures and videos. Even though I have never had
      some big achievements in photography, I enjoy the process of getting a perfect
      shot, including editing and retouching. My favourite kind of photography
      is portrait. Making pictures of people is the most enjoyable for me but,
      also, I like to make pictures of animals and nature. I love when the picture
      is full of bright colours especially green, blue, yellow and red. I have a dream
      to make a photo exhibition and to make a short movie some day."
        id="hobby-5"
        image="test-file-stub"
        name="Photography"
        storageName="photography"
      />
    </div>
  </div>
</div>
`;

exports[`components/Pages/<Home /> should render component with empty props 1`] = `
<div
  className="home-page-container"
>
  <EmptyView
    section="home"
  >
    <p>
      Welcome to my new personal project!
    </p>
    <p>
      Ask questions as you are speaking to me and try to populate the website with the info my bot will provide. Every page will have a progress bar which indicates how much info you have already recovered - so you will know where to stop asking about some particular topic.
    </p>
    <p>
      For example,
       
      <b>
        "
        What is your name?
        "
      </b>
       
      sounds like a good question to start, but it can be something else too, just use your imagination ;)
    </p>
  </EmptyView>
</div>
`;
